### DISCUSSION TITLE
Screen that dumps raw bytes for terminal

### DISCUSSION THREAD

**Discussion by mavam** [ArthurSonzogni/FTXUI#681] [Q&A] [answered]
I've written a TUI that displays structured tabular data. I now need a one-shot rendering opportunity to dump this output anywhere, e.g., to stdout, stderr, file, etc. Where the raw bytes go that make up rendered output is opaque in my API. I can literally just pass bytes downstream.

So what I am looking for is a function that gives me a byte sequence that represents my rendered screen. I would like to offer two options: colored and monochrome. I am not sure where FTXUI internals produce this raw byte sequence before it hits the TTY, but that's where I would hope to get a chance to interpose—optionally enable/disable coloring.

What's the best way to make this happen?

**Comment by mavam** [comment]
My hope was that `ScreenInteractive::ToString` simply dumps the screen as a string, so that I can control the "shipping to the terminal" separately. Unfortunately that's not the case. Is there any side-effect free function to print a screen?

**Comment by ArthurSonzogni** [comment, marked as answer]
Hello @mavam, I think you are describing this function:
```cpp
std::string ToString(Component& component) {
  auto document = component->render();
  auto screen = Screen::Create(Dimension::Full(), Dimension::Fit(document));
  Render(screen, document);
  return screen.ToString();
}
```

**Reply by mavam** [reply]
I *think* this can work. Two questions:

1. ~I'm getting unfortunately no output when I blindly execute your code suggestion. Then I tried adding a `Loop` and called `RunBlockingOnce()`. Then I got the desired output, but it was *also* rendered to the terminal. So this was not quite the right path. Is there another way to "kick the tires" on the component?

2. Since this is a single snapshot of a (container) component, the focus is on the first child of the component. Because the component changes its style when focused, it is highlighted even though I just want a static view without highlighting. If I understand the FTXUI architecture correctly, the user must be somewhere, so the focus must be somewhere. I'm not sure if there's a way to "disable component focus" or whether the usual workaround of `TakeFocus()` called by some other random (potentially invisible) component.

**EDIT**: It works! I accidentally used an `InteractiveScreen` initially. Using the plain `Screen` instead fixed it.

**Reply by mavam** [reply]
When I use `Dimension::Full()` as width, I still get the terminal size as upper bound. My expectation would be that I get the component width instead. Here's an example:

<img width="1496" alt="image" src="https://github.com/ArthurSonzogni/FTXUI/assets/53797/40259316-3a95-427e-b84c-d7e4dcc23fca">

Since such output may be written to a file and users can use horizontal scrolling later on (e.g., `less -S`), I don't want to experience data loss due to cropping.

A workaround of `Dimension::Fixed(some_high_value)` unfortunately (a) does over-provisioning and (b) requires guesswork by the user. I tried switching my `flex` decorators to `flex_grow`, but it
